### PR TITLE
Pro 2744 date and time schema field

### PR DESCRIPTION
**Pull request by ValJed**
[PRO-2744](https://linear.app/apostrophecms/issue/PRO-2744/create-the-dateandtime-schema-field-type)

## Summary
Creates a new `dateAndTime` schema field

## What are the specific steps to test this change?

* Link this banch in an A3 project (testbed)
* Create a field in a module of type `dateAndTime`
* Manipulate this field, it should:
   - Save a date of at least a date is set (if no time it will be 00:00:00)
   - when adding a date, the toggle should automatically pass on on mode.
   - of toggle off no date is saved

## What kind of change does this PR introduce?

- [ ] Bug fix
- [X] New feature
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [X] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [X] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

### CODE REVIEW

File: modules/@apostrophecms/schema/index.js [hunk 1]
```diff
@@ -33,1126 +31,7 @@ module.exports = {
 
     self.enableBrowserData();
 
-    self.addFieldType({
```
Comment (ValJed) [inline modules/@apostrophecms/schema/index.js:36]: this code is untouched, just moved to another file.

File: modules/@apostrophecms/schema/lib/addFieldTypes.js [hunk 2]
```diff
@@ -0,0 +1,1139 @@
...
+
+  self.addFieldType({
+    name: 'time',
+    vueComponent: 'AposInputString',
+    async convert(req, field, data, destination) {
+      destination[field.name] = self.apos.launder.time(data[field.name], field.def);
+    }
+  });
+
+  self.addFieldType({
```
Comment (ValJed) [inline modules/@apostrophecms/schema/lib/addFieldTypes.js:605]: Only this field is new.

File: modules/@apostrophecms/schema/lib/addFieldTypes.js [hunk 3]
```diff
@@ -0,0 +1,1139 @@
...
+    name: 'dateAndTime',
+    vueComponent: 'AposInputDateAndTime',
+    convert (req, field, data, destination) {
+      destination[field.name] = data[field.name]
+        ? self.apos.launder.date(data[field.name], field.def)
```
Comment (ValJed) [inline modules/@apostrophecms/schema/lib/addFieldTypes.js:610]: The launder creates a default date of no value which is weird...

File: CHANGELOG.md [hunk 4]
```diff
@@ -38,9 +44,9 @@
 * Adds possibility for modules to [add extra frontend bundles for scss and js](https://v3.docs.apostrophecms.org/guide/webpack.html). This is useful when the `ui/src` build would otherwise be very large due to code used on rarely accessed pages.
 * Loads the right bundles on the right pages depending on the page template and the loaded widgets. Logged-in users have all the bundles on every page, because they might introduce widgets at any time.
 * Fixes deprecation warnings displayed after running `npm install`, for dependencies that are directly included by this package.
-* Implement custom ETags emission when `etags` cache option is enabled. [See the documentation for more information](https://v3.docs.apostrophecms.org/guide/caching.html).  
-It allows caching of pages and pieces, using a cache invalidation mechanism that takes into account related (and reverse related) document updates, thanks to backlinks mentioned above.  
-Note that for now, only single pages and pieces benefit from the ETags caching system (pages' and pieces' `getOne` REST API route, and regular served pages).
```
Comment (ETLaurent) [inline CHANGELOG.md:43]: Do you have the [EditorConfig](https://marketplace.visualstudio.com/items?itemName=EditorConfig.EditorConfig) extension installed on VS Code?
If so, it should not remove the trailing spaces in markdown files on save (see https://github.com/apostrophecms/apostrophe/blob/main/.editorconfig#L14).

Having a double space at the end of a line makes a line return 🙃

**Pull request merged**